text/0024-cron.md

```diff
@@ -0,0 +1,196 @@
+
+The flags for `qri cron add` are be a subset of those available for `qri update`. Anything `qri update` can do, cron should work the same way: 
+```
+$ qri update me/github_stars --config=org,qri-io,repo,qri-io --secrets=token,$GITHUB_TOKEN
+$ qri cron add me/github_stars --config=org,qri-io,repo,qri-io --secrets=token,$GITHUB_TOKEN
+```
+
+We can see scheduled jobs with `qri cron list`:
+```
+$ qri cron list
+1. b5/dataset_name
+   scheduled: every 8 hours
+   next: Tues. April 11, 11:30:00
+   4 hours from now
+```
+The conventions for `list` should follow all other "list command conventions" for pagination, but listing order should be chronological according to next scheduled update soonest update first, fathest update last.
+
+We can start the qri cron service like this:
+```
+$ qri cron start
+cron update service started. stop it with:
+$ qri cron stop
+```
+
+If the current operating system doesn't yet support a daemonized service, the deamon should start in-process and show a warning message explaining to run this command with some equivalent of `nohup`. We should work to make as few people as possible see that message. As shown in the dialog, the cron daemon must be stoppable with `qri cron stop`.
+
+With cron running Qri will run a lightweight background loop that checks for update events and executes them, logging the results to the event log. To see a log of cron events, run:
+```
+$ qri cron log
+* me/dataset_name
+  succeeded 3h4m ago
+* me/github_stats
+  failed 2d16h27m ago
+  message: error: network timeout
+ # ...
+```
+
+This will show a list of successful & failed job executions
+
+### Scheduling shell scripts
+
+Scheduling datasets directly is the ideal workflow in terms of portability, but a whole set of use cases open by adding the capacity to schedule & execute shell scripts within the same cron environment. While most of this is obviously available by directly using an operating system `cron` service, having scripts for dataset updates collected in one place makes scheduling & maintenance easier. For that reason, shell scripts should be supported as a first-class citizen:
+
+```
+$ qri cron add path/to/update_script.sh R/PT4H/2019-06-01T00:00:00Z
+```
+
+In this case we're case scheduling a cron job to run every 4 hours from now until June 1st, 2019. When qri encounters a file ending in a shell script extension (`.sh` on 'nix platforms), it copies the file into a directory within the fs repo and names the job according the the filename. `qri cron list` now shows both tasks, ordered by which script is the next to execute:
```

Comment:
Hate to be that guy, but any thoughts as to how this will work on Windows? Do we require a cygwin / mingw / msys environment? Do we support powershell? Do we support batch (lol, no, not doing that)?

Reply:
Never don't be that guy. I have literally zero idea, and we should really figure it out. I think what matters here is making sure this kind of thing is _possible_ on windows, even if we don't have a solid plan. As our foremost windows expert, do you think this is doable if we pick a lane in terms of environment?